compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangUserDefinedType.java

```diff
@@ -71,9 +71,18 @@ public void accept(BLangNodeVisitor visitor) {
 
     @Override
     public String toString() {
+        String typeName = this.typeName.value;
+
+        if (typeName.startsWith("$") && this.type != null) {
+            String tName = this.type.toString();
+            if (!tName.equals("other")) {
```

Comment:
When is it "other"? :thinking: Shall we add a comment too?

Reply:
Will add explaining comment.